subworkflows/nf-core/bam_split_by_region/main.nf

```diff
@@ -0,0 +1,64 @@
+//
+// Run SAMtools idxstats, then split reads in input bam into one bam per chromosome in idxstats output
+//
+
+include { SAMTOOLS_VIEW          } from '../../../modules/nf-core/samtools/view/main'
+include { SAMTOOLS_INDEX         } from '../../../modules/nf-core/samtools/index/main'
+
+workflow BAM_SPLIT_BY_REGION {
+
+    take:
+    ch_bam          // channel: [ val(meta), bam , bai ]
+    ch_regions_file // channel: [ regions_file ]
+
+    main:
+
+    ch_versions = Channel.empty()
+
+    // Create channel containing the region names from the bed file.
+    ch_regions = ch_regions_file
+                    .splitCsv ( header: ['seq_name', 'start', 'stop'], sep:'\t' )
+                    .map{
+                        stats ->
+                            // If the regions file contains just a sequence name provide that
+                            if (! stats['start'] ) {
+                                region = stats['seq_name']
+                            } else if ( ! stats['stop']) {
+                            // If a specific position is given, use that
+                                region = [stats['seq_name'], stats['start']].join(":")
+                            } else {
+                            // If a region between specific bps is requested use that
+                                region = [ [stats['seq_name'], stats['start']].join(":"), stats['stop']].join('-')
+                            }
+                        [ region ]
+                    }
```

Comment:
.map{ stats ->
                        // If the regions file contains just a sequence name provide that
                        if (! stats['start'] ) [ stats['seq_name'] ]
                        // If a specific position is given, use that
                        else if ( ! stats['stop']) [ [ stats['seq_name'], stats['start'] ].join(":") ]
                        // If a region between specific bps is requested use that
                        else [ [ [ stats['seq_name'], stats['start'] ].join(":"), stats['stop'] ].join('-') ]
                    }

Reply:
```suggestion
                .map{ stats ->
                    // If the regions file contains just a sequence name provide that
                    if (! stats['start'] ) [ stats['seq_name'] ]
                    // If a specific position is given, use that
                    else if ( ! stats['stop']) [ [ stats['seq_name'], stats['start'] ].join(":") ]
                    // If a region between specific bps is requested use that
                    else [ [ [ stats['seq_name'], stats['start'] ].join(":"), stats['stop'] ].join('-') ]
                }
```